File: backend/Domain/Entities/Author.cs
using Domain.Common;

namespace Domain.Entities;

public class Author : BaseEntity
{
    public string FullName { get; set; } = string.Empty;
    public string? Bio { get; set; }
    public string? Nationality { get; set; }

    public ICollection<BookAuthor> BookAuthors { get; set; } = new List<BookAuthor>();
}

File: frontend/src/types/author.ts
export interface Author {
  id: number;
  fullName: string;
  bio?: string | null;
  nationality?: string | null;
}

export interface CreateAuthorRequest {
  fullName: string;
  bio?: string | null;
  nationality?: string | null;
}

export type UpdateAuthorRequest = CreateAuthorRequest;
